lib/granite/context_proxy.rb

```diff
@@ -0,0 +1,39 @@
+require 'granite/context_proxy/data'
+require 'granite/context_proxy/proxy'
+
+module Granite
+  # This concern contains class methods used for actions and projectors
+  #
+  module ContextProxy
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def using(data)
+        Proxy.new(self, Data.wrap(data))
+      end
+
+      def as(performer)
+        using(performer: performer)
+      end
+
+      def with_context(context)
```

Comment:
What do you think about using [`CurrentAttributes#set`](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html#method-i-set) to implement this functionality?

Reply:
Although it might have unintended consequences...

Reply:
I didn't know about it, but as I understand reading the docs it only allows one global instance of an inheriting class. But we might have many different instances when a parent action initializes a child action with different performer. Conceptually I think it's wrong and it feels to me that we should only allow one context per request and correct the child actions that need different performers. But that would be something that requires a separate big ticket to fix.

Reply:
AFAIK `CurrentAttributes` doesn't work well with jobs, for example.

Reply:
@ojab  hmm good point although https://www.mikeperham.com/2022/07/29/sidekiq-and-request-specific-context/

Reply:
@paneq Wait, why are we using global mutable state (potentially introducing concurrency problem) at all? What's wrong with `.using` creating an anonymous class, as [previously suggested](https://github.com/toptal/granite/pull/76#discussion_r941249579)?

Reply:
We've been using global mutable state and this PR is not changing that. I do agree that ideally we should avoid mutable state and we should investigate possible solutions to that whether it's anonymous class or something else entirely. For now I suggest we keep what we've been using to avoid introducing new problems. The change is already big enough.